clar2wasm/src/words/sequences.rs

```diff
@@ -618,13 +641,19 @@ impl Word for ElementAt {
                 then.i32_const(0);
 
                 // Then push a placeholder for the element type.
-                if let Some(elem_ty) = placeholder_ty {
-                    // Read the element type from the list.
-                    add_placeholder_for_clarity_type(then, &elem_ty)
-                } else {
-                    // The element type is an in-memory type, so we need
-                    // placeholders for offset and length
-                    then.i32_const(0).i32_const(0);
+                match placeholder_ty {
+                    None
+                    | Some(TypeSignature::SequenceType(SequenceSubtype::StringType(
```

Comment:
The use of this optional got a little confusing with this change. I think this could be refactored to be more clear. Maybe another `Enum`, instead of this `Optional`?

Reply:
Done

Reply:
I love this new way! Thanks!